### PR TITLE
ci/cron/check: replace wget with Haskell code

### PR DESCRIPTION
**Pull request by garyverhaegen-da**
As promised in #7696.

CHANGELOG_BEGIN
CHANGELOG_END

### CODE REVIEW

File: ci/cron/src/Main.hs [hunk 1]
```diff
@@ -271,18 +272,25 @@ docs = do
...
-        shell_ $ unlines ["bash -c '",
-            "set -euo pipefail",
-            "eval \"$(dev-env/bin/dade assist)\"",
-            "cd \"" <> tmp <> "\"",
-            "wget --quiet \"" <> show url <> "\"",
-            "'"]))
+              req' <- HTTP.parseRequest (show url)
+              let req = req' { HTTP.requestHeaders = [("User-Agent", "DAML cron (team-daml-language@digitalasset.com)")] }
+              HTTP.withResponse req manager (\resp -> do
+                  let body = HTTP.responseBody resp
```
Comment (cocreature) [inline ci/cron/src/Main.hs:285]: You are basically reimplementing what `http-conduit` gives you. With that, you would end up with something like
```
runResourceT $ HTTP.responseBody resp  $$ sinkFile (tmp </> …)`
```

I don’t particularly mind that so feel free to ignore that suggestion or to address it separately.

File: ci/cron/src/Main.hs [hunk 2]
```diff
@@ -271,18 +272,25 @@ docs = do
...
           (Control.Concurrent.QSem.signalQSem tokens)
           (do
-        shell_ $ unlines ["bash -c '",
-            "set -euo pipefail",
-            "eval \"$(dev-env/bin/dade assist)\"",
-            "cd \"" <> tmp <> "\"",
-            "wget --quiet \"" <> show url <> "\"",
-            "'"]))
+              req' <- HTTP.parseRequest (show url)
+              let req = req' { HTTP.requestHeaders = [("User-Agent", "DAML cron (team-daml-language@digitalasset.com)")] }
```
Comment (cocreature) [inline ci/cron/src/Main.hs:283]: Maybe worth updating that email address but I guess it also doesn’t matter.

**Review by cocreature — APPROVED** [review]
Thanks!

File: ci/cron/src/Main.hs (hunk 1 above)
Comment (garyverhaegen-da) [reply ci/cron/src/Main.hs:285]: I did come upon references to `conduit`, but it seemed to add a lot of concepts I do not know yet. I imagine it's not a lot of complexity once you know them, but this is also not very complex as it stands. I tend to be a bit reluctant to add dependencies for a single function. I'll take a closer look at conduit if it comes up again.

File: ci/cron/src/Main.hs (hunk 2 above)
Comment (garyverhaegen-da) [reply ci/cron/src/Main.hs:283]: What would you suggest instead? AFAIK this one still works and will still reach the relevant people.

**Pull request merged**